data/opam_docs/2.1/03_faq.md

```diff
@@ -0,0 +1,446 @@
+$ opam install --locked <pkg> # use <pkg> locked file, if present
+```
+
+---
+
+#### 🐫  I installed a package by hand / used ``ocamlfind remove`` / fiddled with the installed packages and opam is out of sync. Help!
+
+Don't panic. opam assumes it's controlling what's below `~/.opam/<switch>`, but
+there are several ways you can recover:
+* `opam remove --force` will run the uninstall instructions even if the package
+  is not registered as installed. Then retry `opam install`.
+* `opam reinstall` will try to remove an installed package, but go on to
+  re-installing even if that fails.
+* If all else fails, you can re-install your current set of packages from
+  scratch using `opam switch reinstall`.
+* You can force opam to register an installation or removal _without actually
+  performing anything_ using `opam install|remove --fake`. This is not
+  recommended though, as your manual install may not be exactly equivalent to
+  the one expected by other opam packages, and opam may later on trigger
+  reinstallations or upgrades of the package. Don't complain if you mess up your
+  installation using this! If you want to control how a package is installed or
+  modify it, the right way is `opam pin`.
+* You shouldn't have to, but if you want to restart from scratch, just delete
+  `~/.opam` and get back to `opam init`
+
+---
+
+#### 🐫  What are the minimum requirements?
+
+1GB of memory should be all you need. It was reported that you may run into
+problems with 512MB of RAM and no swap. Of course, compiling the packages may
+need more.
+
+---
+
+#### 🐫  Some packages fail during compilation, complaining about missing dependencies ("m4", "libgtk", etc.)
+
+> NOTE: since opam 2.1.0, the following is directly handled by opam, without
+> relying on a plugin.
+
+They probably depend on system, non-OCaml libraries: they need to be installed
+using your system package manager (apt-get, yum, pacman, homebrew, etc.) since
+they are outside the scope of opam.
+
+opam metadata includes documentation about these external dependencies, on a
+variety of systems/distributions, in the form of a
+[`depexts:`](manual#opamfield-depexts)
+field. Opam should print the required system dependencies, as documented for
+your OS, upon failure, and the `depext` opam plugin can take care of installing
+them for you:
+
+```
+opam depext <opam-packages>
+```
+
+This should install `opam-depext` if needed and prompt to install the system
+packages required by your opam packages or their dependencies, through your OS's
+packaging system.
+
+If that doesn't work...
+* Check for hints printed by the failing package
+* Lookup the development packages corresponding to the error in your system's
+  package repositories.
+* Dependencies for your specific system may not be known, but check the output
+  of `opam list --rec --resolve <package>,<package>... --columns name,depexts:`:
+  it will list dependencies on all known systems and may get you in the right
+  direction.
```

Comment:
```suggestion
  of `opam list --rec --resolve <package>,<package>... --columns name,depexts:`.
  It will list dependencies on all known systems and may get you in the right
  direction.
```